### ISSUE TITLE
From 21.01 -> 21.05: job_working_directory broken - unable to run jobs 🔥

### ISSUE THREAD

**Issue by scholtalbers** [galaxyproject/galaxy#12313]
Updated from 21.01 to 21.05; Jobs all run in the same root of `job_working_directory`, overwriting each other files and due to the setting delete `onsuccess`, files get wiped when one of the jobs finishes.

Possible relevant settings

```
job_working_directory: /g/galaxy/production-data/jobs
cleanup_job: onsuccess
object_store_check_old_style: 'true'   # <- added this setting after getting errors on update (see https://github.com/galaxyproject/galaxy/issues/5339#issuecomment-890917151 )
```
![Screenshot 2021-08-02 at 15 25 33](https://user-images.githubusercontent.com/102411/127869085-248543a5-1c13-4052-967a-6ab469df6ce2.png)

**Comment by scholtalbers** [comment]
Resolved by #12316. FWIW `object_store_check_old_style` shouldn't be used and was not the solution to the initial issue I was facing in https://github.com/galaxyproject/galaxy/issues/5339#issuecomment-890917151

**Comment by mvdbeek** [comment]
Indeed, `object_store_check_old_style` should really never be used in galaxy.yml / galaxy.ini ... if you really need it (this is very old ... you only need this if the datasets in your object store are not nested below a directory) you'd create a new, read-only object store for the old datasets with `object_store_check_old_style: true` in the settings for that object store, and a new object store where you would turn this off.
